### PR TITLE
update DG to reflect Date Storing in Implementations

### PR DESCRIPTION
**Pull request by lawwman**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -345,6 +345,62 @@ image::UndoRedoActivityDiagram.png[width="200"]
 **Cons:** Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things. +
 // end::undoredo[]
 
+// tag::datestoring[]
+=== Date Storing mechanism
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: maybe the 's' in 'storing' can be in lowercase for standardization? Minor issue though.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -345,6 +345,62 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+        this.dateBorrow = new SimpleObjectProperty<>(source.getDateBorrow());
+----
+
+==== Design Considerations
+
+**Aspect:** Implementation of `DateBorrow` +
+**Alternative 1(current choice):** Create a `DateBorrow` class as such but modify the constructor to maintain consistency. +
+**Pros:** Blends well with existing coding style and how the commands work. +
+**Cons:** As of now there is no problems with the implementation. +
+**Alternative 2:** Have the User manually key in the date. +
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: user*

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -345,6 +345,62 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+**Pros:** Implementation is very simple as the developer could just follow current coding style to create a new field. +
+**Cons:** Is not as user friendly. As of now, there amount of fields is rather high and having to input another one is tedious for the users.
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: - 'Is not as user friendly' could be rephrased and expressed in a proper sentence. 
- user-friendly*
- The second sentence needs to be rephrased too.

**Review by jelneo — CHANGES_REQUESTED** [review]
Please refer to the comments. Otherwise, good job! Clear explanation for the mechanism

**Review by jelneo — APPROVED** [review]

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -345,6 +345,64 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+This is because the above mentioned constructor was meant to make a copy of the `Person` with a given source. Hence the following line was added to ensure consistency.
+[source,java]
+----
+        this.dateBorrow = new SimpleObjectProperty<>(source.getDateBorrow());
+----
+
+==== Design Considerations
+
+**Aspect:** Implementation of `DateBorrow` +
+**Alternative 1(current choice):** Create a `DateBorrow` class as such but modify the constructor to maintain consistency. +
```
Comment (khooroko) [inline docs/DeveloperGuide.adoc:397]: perhaps you want to rephrase this to include how it uses the current date, since this choice and alternative 2 don't seem mutually exclusive

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -345,6 +345,64 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+    private ObjectProperty<Phone> phone;
+    private ObjectProperty<Email> email;
+    private ObjectProperty<Address> address;
+    private ObjectProperty<PostalCode> postalCode;
+    private ObjectProperty<Cluster> cluster;
+    private ObjectProperty<Debt> debt;
+    private ObjectProperty<Interest> interest;
+    private ObjectProperty<DateBorrow> dateBorrow;
+----
+Such an implementation doesn't allow for errors when creating the field as there is no room for mistakes on the user's side. When the `Person` instance is created, the following line is called: +
```
Comment (khooroko) [inline docs/DeveloperGuide.adoc:367]: does not*. don't use contractions in documents
